tests/test_dataManager.py

```diff
@@ -0,0 +1,120 @@
+# This file is part of ts_FiberSpectrograph.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os.path
+import tempfile
+import unittest
+
+import astropy.time
+import numpy as np
+
+from lsst.ts.FiberSpectrograph import SpectrographData, DataManager
+
+
+class TestAvsFiberSpectrograph(unittest.TestCase):
+    """Tests of the data management API for the fiber spectrograph CSC.
+    """
+    def setUp(self):
+        self.telescope = "TestMain"
+        self.instrument = "TestBlue"
+        self.n_pixels = 1234
+        self.wavelength = np.arange(300, 1000, 1000, dtype=np.float64)
+        self.spectrum = np.random.random(1000)
+        self.duration = 2.5
+        self.time_str = '1999-01-01T00:00:00.000'
+        self.date_begin = astropy.time.Time(self.time_str, scale='tai')
+        self.date_end = self.date_begin + astropy.time.TimeDelta(self.duration, format='sec')
+        self.temperature = -273
+        self.temperature_setpoint = -274
+        self.type = "totally real data"
+        self.source = "blacklight"
+        self.data = SpectrographData(wavelength=self.wavelength,
+                                     spectrum=self.spectrum,
+                                     duration=self.duration,
+                                     date_begin=self.date_begin,
+                                     date_end=self.date_end,
+                                     type=self.type,
+                                     source=self.source,
+                                     temperature=self.temperature,
+                                     temperature_setpoint=self.temperature_setpoint,
+                                     n_pixels=self.n_pixels)
+
+    def check_header(self, header):
+        self.assertEqual(header['TELESCOP'], self.telescope)
```

Comment:
If the header compares as a dict then you can set up the output dict once and then use `self.assertEqual(header, self.reference_header)`  Depends whether you are worried about extra headers turning up that you aren't expecting.

Reply:
It doesn't look like dict-like comparison works for astropy Headers: the header is being compared as a big FITS character block instead of a dict.

Reply:
Also, we can't do a direct comparison, because there will be added keys in the final output FITS header (e.g. 'SIMPLE').

Reply:
I created a dict in the class and iterated over its `items()`.